internal/plugin6/grpc_provider.go

```diff
@@ -1515,11 +1515,8 @@ func (p *GRPCProvider) ReadStateBytes(r providers.ReadStateBytesRequest) (resp p
 	for {
 		chunk, err := client.Recv()
 		if err == io.EOF {
-			// End of stream, we're done
-			if chunk != nil {
-				// TODO: The EOF error could be just returned alongside the last chunk?
-				resp.Diagnostics = resp.Diagnostics.Append(convert.ProtoToDiagnostics(chunk.Diagnostics))
-			}
+			// No chunk is returned alongside an EOF.
+			// And as EOF is a sentinel error it isn't wrapped as a diagnostic.
 			break
 		}
```

Comment:
Please see this comment in our terraform-plugin-go PR explaining why we will never see a chunk alongside the EOF error here: https://github.com/hashicorp/terraform-plugin-go/pull/563/files#r2362735975